paddle/phi/kernels/onednn/matmul_kernel.cc

```diff
@@ -0,0 +1,450 @@
+
+    OT *output_data = output->mutable_data<OT>(ctx.GetPlace(), buffer_size);
+    output->set_mem_desc(dst_desc);
+    return memory(dst_desc, engine_, phi::funcs::to_void_cast<OT>(output_data));
+  }
+
+  memory Reorder(const memory::desc &src_desc,
+                 const memory::desc &dst_desc,
+                 void *src_data,
+                 void *dst_data = NULL) {
+    auto src_mem = memory(src_desc, engine_, src_data);
+    auto dst_mem = dst_data ? memory(dst_desc, engine_, dst_data)
+                            : memory(dst_desc, engine_);
+
+    auto reorder = dnnl::reorder(src_mem, dst_mem);
+
+    auto &astream = OneDNNContext::tls().get_stream();
+    {
+      paddle::platform::RecordEvent record_reorder(
+          "int_reorder",
+          paddle::platform::TracerEventType::UserDefined,
+          2,
+          paddle::platform::EventRole::kUniqueOp);
+      reorder.execute(astream, src_mem, dst_mem);
+      astream.wait();
+    }
+
+    return dst_mem;
+  }
+
+  memory TransposeInputY(const DenseTensor *input_y) {
+    auto dims = phi::vectorize<int64_t>(input_y->dims());
+    std::swap(dims[0], dims[1]);  // Correct output dimensions
+    auto src_desc =
+        CreateMemDescriptor<YT>(dims, funcs::OneDNNMemoryFormat::io);
+    auto dst_desc =
+        CreateMemDescriptor<YT>(dims, funcs::OneDNNMemoryFormat::oi);
+    return Reorder(
+        src_desc, dst_desc, phi::funcs::to_void_cast<YT>(input_y->data<YT>()));
+  }
+
+  const dnnl::engine &engine_;
+  paddle::optional<memory> x_input_;
+  paddle::optional<memory> y_input_;
+  paddle::optional<memory> output_;
+  paddle::optional<inner_product_forward> mul_;
+  static constexpr bool is_int8_ = funcs::is_int8<XT>();
+};
+
+/* OT: output data type */
+template <typename XT, typename YT, typename OT>
+std::shared_ptr<MulPrimitiveFactory<XT, YT, OT>> GetPrimitiveFactory(
+    const OneDNNContext &dev_ctx,
+    const DenseTensor *input_x,
+    const DenseTensor *input_y,
+    const dnnl::engine &onednn_engine) {
+  std::string key =
+      funcs::CreateKey(dev_ctx,
+                       paddle::framework::TransToProtoVarType(input_x->dtype()),
+                       phi::vectorize(input_x->dims()),
+                       paddle::framework::TransToProtoVarType(input_y->dtype()),
+                       phi::vectorize(input_y->dims()),
+                       dev_ctx.GetOutputsName("Out")[0]);
+  key = funcs::ExtendKeyWithThreadInfoIfNeeded(dev_ctx, key);
+
+  auto prim_creator = std::static_pointer_cast<MulPrimitiveFactory<XT, YT, OT>>(
+      dev_ctx.GetBlob(key));
+
+  if (prim_creator == nullptr) {
+    prim_creator =
+        std::make_shared<MulPrimitiveFactory<XT, YT, OT>>(onednn_engine);
+    dev_ctx.SetBlob(key, prim_creator);
+  }
+
+  return prim_creator;
+}
+
+/* XT: input x data type, YT: input y data type */
+template <typename XT, typename YT, typename Context>
+inner_product_forward GetMulPrimitive(const Context &dev_ctx,
```

Comment:
```suggestion
template <typename XT, typename YT>
inner_product_forward GetMulPrimitive(const Context &dev_ctx,
```
`OneDNNContext` is already passed as argument